### PR TITLE
Add gallery of examples to documentation

### PR DESCRIPTION
**Pull request by adrn**
*Work in progress*

This is the result of [discussions at Python in Astronomy 2016](https://docs.google.com/document/d/1oY9iuMiWp15XCYCx5PEUnPGD5W3ge9X8aiqxofiWVHw/edit) about the state of [astropy-tutorials](https://github.com/astropy/astropy-tutorials). We decided to implement a trial version of a [sphinx-gallery](https://github.com/sphinx-gallery/sphinx-gallery) of examples in the astropy documentation similar to the scikit-learn [examples gallery](http://scikit-learn.org/stable/auto_examples/index.html). 

[Here's a live demo of the gallery in the astropy docs](http://adrian.pw/astropy/auto_examples/index.html)

Outstanding issues / tasks:

- [x] organize the examples by subpackage
- [ ] look through the astropy documentation and see if anything should be pulled out into the example gallery:
    - [x] http://docs.astropy.org/en/stable/io/fits/appendix/faq.html#how-do-i-create-a-multi-extension-fits-file-from-scratch
    - [x] http://docs.astropy.org/en/stable/io/fits/appendix/faq.html#how-do-i-open-a-very-large-image-that-won-t-fit-in-memory
    - [x] http://docs.astropy.org/en/stable/coordinates/observing-example.html
    - [x] http://docs.astropy.org/en/stable/coordinates/sgr-example.html
    - [ ] One of the simple model-fitting examples from astropy.modeling main page

Some strange issues with sphinx-gallery:

- [x] There seems to be extra indentation for the first line of code when output has multiple lines (see: sphinx-gallery/sphinx-gallery#109 (~~opened~~ merged a PR to fix: sphinx-gallery/sphinx-gallery#111) 
- [x] ~~Mouseover for gallery thumbnails doesn't always show the name of the example (not going to be fixed before this PR is merged)~~ (seems to be resolved)
- [x] There doesn't seem to be a way to choose the thumbnail image for an example that generates multiple figures sphinx-gallery/sphinx-gallery#110 (~~opened~~ merged a PR to fix: sphinx-gallery/sphinx-gallery#114)

TODO items to be handled after this PR:

- When sphinx-gallery 0.1.3 or 0.2 is released, use the new thumbnail image selection feature to show the 2nd plot from the observing example.
- Replace the default sphinx-gallery placeholder image (this requires adding the functionality to sphinx-gallery)
- Some functions resolve automatically to API links, some don't...This is a known issue and needs to be fixed in sphinx-gallery (sphinx-gallery/sphinx-gallery#113) 

cc @kelle

### CODE REVIEW

**Comment by Cadair** [comment]
Have you tested this on readthedocs? I am having memory consumption issues on their build service.

**Comment by adrn** [comment]
@Cadair I haven't, but we also haven't asked those nice people yet for more resources

**Comment by adrn** [comment]
BTW: to anyone following, my thesis is due next week so I haven't worked on this since getting back from PyAstro16. It's not far from being complete -- there are a few remaining issues for me to solve / submit PR's to sphinx-gallery itself but those shouldn't take too much time.

**Comment by astrofrog** [comment]
Dr @adrn (congrats! 🎉 ) - do you think it would be possible to finish this up by the end of this week so that we can include it in the 1.2 docs? (feature freeze is this Friday)

**Comment by adrn** [comment]
Thanks @astrofrog! 

Yea, totally -- I'm back at my desk today.

**Comment by adrn** [comment]
OK, I think this is almost ready for final review. I ran out of time to write a model fitting example but maybe @pllim or @nden would be willing to help with that once this PR is merged?

**Comment by pllim** [comment]
My feeling from PyAstro16 is that people don't like Astropy fitters and want to use Sherpa fitters (or something like that). I thought someone wrote an example like that using MCMC during PyAstro16, no? I cannot seem to find it in the Google doc but I remember hearing about it. That would be a good example to include here.

**Comment by adrn** [comment]
Hm, I thought @Cadair's PR to astropy-helpers to check for "build succeeded" got merged? cc @eteq

**Comment by bsipocz** [comment]
@adrn - But there wasn't any new helpers release yet.

**Comment by adrn** [comment]
Oh right - sorry I thought I remembered seeing a 1.1.3 release. oops!

**Comment by adrn** [comment]
In that case, I think this is ready for review. cc @astrofrog @kelle

**Comment by astrofrog** [comment]
@adrn - could you rebase this and add a changelog entry?

File: docs/index.rst [hunk 1]
```diff
@@ -58,6 +58,7 @@ User Documentation
    overview
    install
    getting_started
+   auto_examples/index
```
Comment (astrofrog) [inline docs/index.rst]: If you want to put it here, it should only add one entry to the TOC called 'Tutorials' or 'Gallery of Tutorials'. At the moment it adds two obscure ``astropy.coordinates`` and ``astropy.io`` entries.

File: examples/coordinates/plot_obs-planning.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+"""
+===========================
+Plot the AltAz of an object
```
Comment (astrofrog) [inline examples/coordinates/plot_obs-planning.py]: Determining and plotting the altitude/azimuth of a celestial object

**Comment by taldcroft** [comment]
@adrn - just had a look at this.  Awesome work!  I really hope this will open the floodgates for user-contributed examples now.  It makes it really easy to add examples and makes the results highly visible in a nice format.

**Comment by adrn** [comment]
@astrofrog RE: changelog, should I add to "Other changes and additions" or make a new category for documentation changes?

@taldcroft Thanks! Yea, I'm really hoping this makes examples more visible and easier for people to contribute.

**Comment by astrofrog** [comment]
@adrn - 'other changes' is fine, I can see if there is enough to make a new section when I tidy up the changelog

**Comment by eteq** [comment]
@adrn - can you try just adding a commit at the bottom of this PR to update the helpers submodule to the latest helpers?  I *think* that will fix the build_succeeded issue (astropy/astropy-helpers#223) without forcing us to do a helpers release now just for that.

**Comment by eteq** [comment]
(To check this I just started a quick run on my travis: https://travis-ci.org/eteq/astropy/builds/130121818)

**Comment by eteq** [comment]
Note that it all worked on my branch where I just added one more "update the helpers to master" commit: https://travis-ci.org/eteq/astropy/builds/130121818 .

@adrn, if you can get to this now (and the changelog entry) I'm happy with merging once those are in and not waiting for this to run.

**Comment by eteq** [comment]
Alright, this is now merged in 660b91f733d8e2364d691eecb33eb31ca187f766 !  (github doesn't recognize it as "merged" because I had to rebase on master... but I'm closing this because it's in, I promise.)  Nice work @adrn and @kelle!

**Comment by astrofrog** [comment]
Thanks @adrn and @kelle!

**Comment by adrn** [comment]
Thanks @eteq for finishing up the final steps (I was busy last night) and thanks @astrofrog for reviewing!

**Pull request closed without merging**